### ISSUE TITLE
Add ignoreDuplicateBuildFailure flag to specifically ignore failure from duplicate builds

### ISSUE THREAD

**Issue by olsh** [GoogleChrome/lighthouse-ci#219]
When `lhci upload` command is called on the same commit twice, the server returns an error
```
Error: Unexpected status code 422
  {"message":"Build already exists for hash \"1154a961627df3acf0ba440e0fa3a19017183dfc\""}
    at ApiClient._convertFetchResponseToReturnValue 
```

It would be nice to have an ability to either ignore this or override existing build on the LightHouse server.

It can be useful if you want to trigger the same build on the CI server twice. For example, on the first run, some steps (after LightHouse) are failed. You fixed the issues and run the build again with the same code.

**Comment by patrickhulce** [comment]
Thanks for filing @olsh! We want to prevent loss of data by default especially while the server is unauthenticated (current token can be made public and is additive only). We're planning on introducing some sort of admin token concept that would let you do things like overwrite and delete historical data.

It does make sense though to more broadly allow overwriting of just the most recent commit for a particular branch in retry scenarios. Would you be willing to change a setting in your lighthouserc.json where you host your server to enable something like `allowOverwriteOfLatestBranchBuild: true` to turn this behavior on? That might be a reasonable compromise.

**Comment by olsh** [comment]
> Would you be willing to change a setting in your lighthouserc.json where you host your server to enable something like allowOverwriteOfLatestBranchBuild: true to turn this behavior on?

Yes, I think this would work. 

There is a more safe approach to this issue: to add a switch in the client config something like `ignoreIfBuildExists: true`. In this case, the client exit with code 0 (without overriding the latest build on the server) even if the build already exists on the server.

**Comment by patrickhulce** [comment]
> There is a more safe approach to this issue: to add a switch in the client config something like ignoreIfBuildExists: true. In this case, the client exit with code 0 (without overriding the latest build on the server) even if the build already exists on the server.

This is essentially the default behavior of `autorun` already which ignores the exit code of upload. If you are invoking `lhci upload` manually, then it's encouraged to also not fail the build on failed uploads (e.g. documentation examples place the upload bit in `after_success:` of travis). An explicit opt-in flag for this one error seems reasonable though since unlike other scenarios, this will never be resolved by retrying 👍 

Mind if this issue tracks `ignoreDuplicateBuildFailure` and let #85 continue to track the management aspects blocked by additional auth?

**Comment by olsh** [comment]
> This is essentially the default behavior of autorun already which ignores the exit code of upload.

Whoops, missed this, thanks! 👍 

> Mind if this issue tracks ignoreDuplicateBuildFailure and let #85 continue to track the management aspects blocked by additional auth?

Yes, sure.

Thank you for your help, @patrickhulce!
